src/python/WMCore/Credential/Proxy.py

```diff
@@ -309,6 +309,42 @@ def getUserName(self, proxy=None):
 
         return uName
 
+    def getMyproxyUsernameForCRAB(self):
+        """
+        :return: username (a string) to be used when myproxy-* command is called with the option
+               -l | --username        <username> Username for the delegated proxy
+
+        CRAB (and only CRAB as of March 2020) needs to obtain from myproxy a proxy from a credential
+        uploaded by another user. So that it can submit jobs and data transfers authenticated
+        with the user proxy. For this a special credential is uploaded in myproxy server by
+        CRAB Client which is connected to a username associated to the user and has a
+        list of authorized retrievers as a list of DN's of CRAB TaskWorkers maintained in
+        central CRAB configuration.
+        The username is passed to myproxy-* command via the "-l" option
+        Two different algorithms for defining this username have been used in CRAB:
+        1. the hash of the user DN + the fqdn of the CRAB REST host
+        2. the user CERN primary account username + the _CRAB string
+        During spring 2020 CRAB migrates from 1. to 2. For a smooth migration the
+        new client needs to upload both credentials and the TW will try 2. and fall back to 1.
+        Only after all tasks submited with old client are gone from the system, can we change
+        to support only 2.
+        The reasons to change from 1. to 2. are to make the username readable (helps support) and
+        to allow using different REST hosts (helps in K8s world). The reasons for the complicated
+        recipe in 1. are unknown, aside some security by obscurity attempt.
+        The caller decides if the call to myproxy-* done by this module will use 1. or 2. via
+        the userName key in the dictionary passed as argument to Proxy() at __init__ time
+           - if the dictionary contains the key 'userName', algorithm 2 is used
+           - if the dictionary does not have it, algorithm 1. is used
+        """
+        if self.userName:
+            self.logger.debug("using %s as credential login name", self.userName)
+            username = self.userName
+        else:
+            self.logger.debug(
+                "Calculating hash of %s for credential name" % (self.userDN + "_" + self.myproxyAccount))
```

Comment:
I don't remember which pylint complaint is triggered here. A better logging format would be:
```
"Calculating hash of %s for credential name", self.userDN + "_" + self.myproxyAccount)
```

No need to update this patch for this line only though.

Reply:
done